### PR TITLE
fix: tier handlers must write error only once

### PR DESCRIPTION
**Pull request by jiuker**
## Description

https://github.com/minio/minio/blob/a7f266c9075a8c2eee5fec39dac0b5411763d2a4/cmd/admin-handler-utils.go#L37-L61
It looks like this exception has been handled here.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

**Comment by harshavardhana** [comment]
We need to make sure we return error for `globalTierConfigMgr` is nil.

**Comment by jiuker** [comment]
@harshavardhana /cc now?

File: cmd/tier-handlers.go [hunk 1]
```diff
@@ -133,6 +137,15 @@ func (api adminAPIHandlers) ListTierHandler(w http.ResponseWriter, r *http.Reque
...
 
+	if globalTierConfigMgr == nil {
+		writeErrorResponseJSON(ctx, w, errorCodes.ToAPIErr(ErrServerNotInitialized), r.URL)
+		return
+	}
+
+	if globalTierConfigMgr == nil {
+		writeErrorResponseJSON(ctx, w, errorCodes.ToAPIErr(ErrServerNotInitialized), r.URL)
+		return
+	}
```
Comment (harshavardhana) [inline cmd/tier-handlers.go]: Looks like copy paste error?

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Review by harshavardhana — APPROVED** [review]

File: cmd/tier-handlers.go [hunk 2]
```diff
@@ -202,11 +211,14 @@ func (api adminAPIHandlers) RemoveTierHandler(w http.ResponseWriter, r *http.Req
 
 	defer logger.AuditLog(ctx, w, r, mustGetClaimsFromToken(r))
 
-	objAPI, _ := validateAdminReq(ctx, w, r, iampolicy.SetTierAction)
-	if objAPI == nil || globalNotificationSys == nil || globalTierConfigMgr == nil {
+	if globalTierConfigMgr == nil {
```
Comment (krisis) [inline cmd/tier-handlers.go]: `globalTierConfigMgr` is [initialized](https://github.com/minio/minio/blob/master/cmd/server-main.go#L366) to a non-nil value before the object layer is [initialized](https://github.com/minio/minio/blob/master/cmd/server-main.go#L416). So, we don't need to check `if globalTierConfigMgr == nil` in any of the tier handlers. I realized this only on reviewing this PR.

**Review by krisis — COMMENTED** [review]
Good catch!

**Review by krisis — APPROVED** [review]

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**